### PR TITLE
Patch delete

### LINKED ISSUE #25: Add endpoints for Items
blocked by #23 
add endpoints for updating and deleting individual `Items`. They already can be created and read through the existing `/wishlist/` endpoints.

- [x] `PATCH /item/:id`
- [x] `DELETE /item/:id`

`PATCH /item/:id`  should only update the `name` attribute of an item since the others are IDs that are managed by the database.

### PR DESCRIPTION
**Pull request by TylerSowers**

### CODE REVIEW

File: server/app/app.js [hunk 1]
```diff
@@ -49,6 +49,17 @@ const updateWishlist = async (id, { name, author }) => {
   });
 };
 
+const updateItem = async (id, { name }) => {
+  return new WishListItem({ id, name }).save(null, {
+    require: true,
+    method: 'update',
+  });
```
Comment (thirionlogan) [inline server/app/app.js:56]: This should include `patch: true`. Refer to the [Bookshelf documentation for save](https://bookshelfjs.org/api.html#Model-instance-save)

File: server/app/app.js [hunk 2]
```diff
@@ -103,4 +114,46 @@ app.put('/wishlist/:id', (req, res) => {
...
+
+
+
+//Delete to delete specific items not entire wishlist's
+
+
+
+
+
+
```
Comment (thirionlogan) [inline server/app/app.js:159]: remove these comments

File: server/app/app.test.js [hunk 3]
```diff
@@ -169,4 +170,42 @@ describe('Endpoints', () => {
...
+      const response = await request(app).delete('/item/500');
+      expect(response.statusCode).toBe(500)
+    })
+  });
 });
+
+
+
+
+
```
Comment (thirionlogan) [inline server/app/app.test.js:211]: What's with the extra white space here?

File: server/app/app.test.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 const request = require('supertest');
 const app = require('./app');
 const db = require('../data/db');
+const { update } = require('../data/db');
```
Comment (thirionlogan) [inline server/app/app.test.js:4]: Why is this import added?

**Review by thirionlogan — CHANGES_REQUESTED** [review]
The `updateItem` function would overwrite the entire entry in the database instead of only modifying the props passed to it without specifying the `patch` option.

File: server/app/app.test.js (hunk 4 above)
Comment (thirionlogan) [reply server/app/app.test.js:4]: It isn't used, so it should probably be removed

File: server/app/app.js (hunk 1 above)
Comment (thirionlogan) [reply server/app/app.js:56]: The relevant part of the documentation:
> - [patch=false] Boolean
Only save attributes supplied as arguments to the save call, ignoring any attributes that may be already set on the model.

**Review by thirionlogan — APPROVED** [review]
Good Job!

**Review by ghost — APPROVED** [review]
